tests/infer/test_integrator.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import torch
+from torch.autograd import Variable
+
+from pyro.infer.mcmc.verlet_integrator import verlet_integrator
+from tests.common import assert_equal
+
+
+def test_harmonic_oscillator():
+
+    def energy(q, p):
+        return 0.5 * p['x']**2 + 0.5 * q['x']**2
+
+    def grad(q):
+        return {'x': q['x']}
+
+    q = {'x': Variable(torch.Tensor([0.0]), requires_grad=True)}
+    p = {'x': Variable(torch.Tensor([1.0]), requires_grad=True)}
+    energy_cur = energy(q, p)
+    print("Energy - current: {}".format(energy_cur.data[0]))
+    q_new, p_new = verlet_integrator(q, p, grad, 0.01, 100)
+    assert q_new['x'].data[0] != q['x'].data[0]
+    energy_new = energy(q_new, p_new)
+    assert_equal(energy_new, energy_cur)
+    assert_equal(q_new['x'].data[0], np.sin(1.0), prec=1.0e-4)
+    assert_equal(p_new['x'].data[0], np.cos(1.0), prec=1.0e-4)
+    print("q_old: {}, p_old: {}".format(q['x'].data[0], p['x'].data[0]))
+    print("q_new: {}, p_new: {}".format(q_new['x'].data[0], p_new['x'].data[0]))
+    print("Energy - new: {}".format(energy_new.data[0]))
+    print("-------------------------------------")
+
+
+def test_circular_planetary_motion():
```

Comment:
Thanks for adding these tests in, @martinjankowiak! I have moved them to test_integrator.py. Please feel free to modify/tune them further if needed.